### ISSUE TITLE
per_horizon_rmse overflows float32 during intermediate squaring, reports inf for finite errors

### ISSUE THREAD

**Issue by Svector-anu** [SlopDotCash/asi#263]
## What's wrong

\`per_horizon_rmse\` squared float32 errors directly before averaging:

\`\`\`python
sq_err = (predictions - forward_returns) ** 2
return jnp.sqrt(jnp.mean(sq_err, axis=0))
\`\`\`

For a large but finite error (e.g. \`2.0e20\`, well within float32's representable magnitude of ~3.4e38), squaring it first (\`(2e20)^2 = 4e40\`) overflows float32 during the intermediate step, even though the correct final RMSE is itself finite and representable. This silently corrupts the nexting measurement harness by reporting infinity for a genuinely finite error magnitude.

Reproduced directly before any fix, checked against an independent float64 reference:

\`\`\`text
actual: [inf] finite: [False]
float64 reference: [2.00000004e+20] finite: [ True]
reference representable in float32: [ True]
\`\`\`

## Reachability (honest)

\`per_horizon_rmse\` is directly importable as a public module-level function (\`from alberta_framework.utils.nexting import per_horizon_rmse\`), reachable with real caller-controlled float32 inputs. Neither \`alberta_framework/__init__.py\` nor \`alberta_framework/utils/__init__.py\` re-exports it (\`multi_channel_horizon_returns\` from the same module is re-exported, but \`per_horizon_rmse\` is not), so this is not currently producing wrong numbers in a shipped package-root-exported pipeline — it's a real gap in a public module-level evaluation API.

## Fix

The nexting module cites Modayil, White, and Sutton (2014) for the evaluation protocol; this fix doesn't alter that protocol or its forward-view formula, it only evaluates the same documented RMSE definition using the algebraically equivalent scaled form \`max(abs(error)) * sqrt(mean((error / max(abs(error)))**2))\` for fully finite columns, avoiding the overflow-prone intermediate square. Columns containing NaN or infinity retain the original direct calculation so existing diagnostic propagation stays visible (a genuinely non-finite input should still report non-finite, this fix only closes the *spurious* overflow on finite-but-large inputs).

PR incoming with a regression test using two float32 errors of \`2.0e20\` (whose direct squares overflow float32, though the correct RMSE remains \`2.0e20\`), plus a mutation-resistance transcript.
